src/duplicacy_backupmanager.go

```diff
@@ -544,6 +547,13 @@ func (manager *BackupManager) Backup(top string, quickMode bool, threads int, ta
 				entry.Size = fileSize
 				uploadedEntries = append(uploadedEntries, entry)
 
+				// Did the file actually change?
+				if entry.RemoteEntry != nil && entry.RemoteEntry.Hash == entry.Hash && entry.RemoteEntry.Size == entry.Size {
```

Comment:
This is wrong.  Note that the local hash was just copied from the remote hash previously (on line 324) so the two hashes are always equal.

Reply:
So from what I've understood (entirely possible I've missed something else) at this point entry.Size and entry.Hash have been updated to the file's current hash and size, as calculated by the chunkmaker. The entry's RemoteEntry however is the entry as found from the previous snapshot, and so contains the file's hash and size as they were in the previous backup. So they can be different if the file has changed, and testing this locally shows that the hashes are indeed different when the file has been changed.

But now you've pointed out that the entry's hash is originally set to the previous hash before this function is called I realise that I don't need the RemoteEntry at all and can just compare entry.Hash to the new hash before overwriting it in line 546. That seems a lot cleaner.